id: PYSEC-0000-CVE-2024-41942
severity:
- type: CVSS_V3
  score: CVSS:3.1/AV:N/AC:L/PR:H/UI:N/S:U/C:H/I:H/A:H
details: |-
  JupyterHub is software that allows one to create a multi-user server for Jupyter notebooks. Prior to versions 4.1.6 and 5.1.0, if a user is granted the `admin:users` scope, they may escalate their own privileges by making themselves a full admin user. The impact is relatively small in that `admin:users` is already an extremely privileged scope only granted to trusted users.
  In effect, `admin:users` is equivalent to `admin=True`, which is not intended. Note that the change here only prevents escalation to the built-in JupyterHub admin role that has unrestricted permissions. It does not prevent users with e.g. `groups` permissions from granting themselves or other users permissions via group membership, which is intentional. Versions 4.1.6 and 5.1.0 fix this issue.
affected:
- package:
    name: jupyterhub
    ecosystem: PyPI
    purl: pkg:pypi/jupyterhub
  ranges:
  - type: GIT
    repo: https://github.com/jupyterhub/jupyterhub
    events:
    - introduced: "0"
    - fixed: 99e2720b0fc626cbeeca3c6337f917fdacfaa428
    - fixed: ff2db557a85b6980f90c3158634bf924063ab8ba
  - type: ECOSYSTEM
    events:
    - introduced: "0"
    - fixed: 4.1.6
references:
- type: ADVISORY
  url: https://github.com/jupyterhub/jupyterhub/security/advisories/GHSA-9x4q-3gxw-849f
- type: FIX
  url: https://github.com/jupyterhub/jupyterhub/commit/99e2720b0fc626cbeeca3c6337f917fdacfaa428
- type: FIX
  url: https://github.com/jupyterhub/jupyterhub/commit/ff2db557a85b6980f90c3158634bf924063ab8ba
aliases:
- CVE-2024-41942
- GHSA-9x4q-3gxw-849f
modified: "2024-08-12T15:53:27Z"
published: "2024-08-08T15:15:17Z"
